base/linalg/cholesky.jl

```diff
@@ -324,11 +324,13 @@ convert{T}(::Type{CholeskyPivoted{T}},C::CholeskyPivoted) =
     CholeskyPivoted(AbstractMatrix{T}(C.factors),C.uplo,C.piv,C.rank,C.tol,C.info)
 convert{T}(::Type{Factorization{T}}, C::CholeskyPivoted) = convert(CholeskyPivoted{T}, C)
 
-full{T,S}(C::Cholesky{T,S}) = C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]'
-function full(F::CholeskyPivoted)
-    ip = invperm(F[:p])
-    return (F[:L] * F[:U])[ip,ip]
-end
+convert(::Type{Matrix}, C::Cholesky) = convert(Array, C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]')
+convert(::Type{Array}, C::Cholesky) = convert(Matrix, C)
+full(C::Cholesky) = convert(Array, C)
+
+convert(::Type{Matrix}, F::CholeskyPivoted) = (ip = invperm(F[:p]); convert(Array, (F[:L] * F[:U])[ip,ip]))
```

Comment:
it's still multiple statements, assigning a temporary variable then using it - worth doing those on separate lines IMO